File: 590/NAryTreePostorderTraversal.java
/*
https://leetcode.com/problems/n-ary-tree-postorder-traversal/description/?envType=daily-question&envId=2024-08-26
*/

/*
// Definition for a Node.
class Node {
    public int val;
    public List<Node> children;

    public Node() {}

    public Node(int _val) {
        val = _val;
    }

    public Node(int _val, List<Node> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {

    public static void depthFirstSearch_forTrees(Node root, List<Integer>resultantPostOrderTraversal){
        if(root==null) return;
        List<Node>currentChildren=root.children;
        int sizeOfChildren=currentChildren.size();
        for(int i=0; i<sizeOfChildren; ++i){
            Node currentChild=currentChildren.get(i);
            if(currentChild!=null){
                depthFirstSearch_forTrees(currentChild, resultantPostOrderTraversal);
            }
        }
        resultantPostOrderTraversal.add(root.val);
        return;
    }

    public List<Integer> postorder(Node root) {
        List<Integer>resultantPostOrderTraversal=new ArrayList<>();
        depthFirstSearch_forTrees(root, resultantPostOrderTraversal);
        return resultantPostOrderTraversal;
    }
}

File: 5/590/NAryTreePostorderTraversal.cpp
/*
https://leetcode.com/problems/n-ary-tree-postorder-traversal/description/?envType=daily-question&envId=2024-08-26
*/

/*
// Definition for a Node.
class Node {
public:
    int val;
    vector<Node*> children;

    Node() {}

    Node(int _val) {
        val = _val;
    }

    Node(int _val, vector<Node*> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
public:

    void dfs(Node *root, vector<int>&res){
        if(root==nullptr) return;
        for(auto &child: root->children){
            if(child!=nullptr){
                dfs(child, res);
            }
        }
        res.push_back(root->val);
        return;
    }

    vector<int> postorder(Node* root) {
        vector<int>res;
        dfs(root, res);
        return res;
    }
};